include/na/zoned/layout_synthesizer/PlaceAndRouteSynthesizer.hpp

```diff
@@ -0,0 +1,133 @@
+   */
+  struct Config {
+    /// Configuration for the placer
+    typename Placer::Config placerConfig{};
+    /// Configuration for the router
+    typename Router::Config routerConfig{};
+    NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT(Config, placerConfig,
+                                                routerConfig);
+  };
+  /**
+   * Collection of statistics collected during the synthesis process for the
+   * placement and routing synthesizer.
+   */
+  struct Statistics {
+    int64_t placementTime; ///< Time taken for placement in us
+    int64_t routingTime;   ///< Time taken for routing in us
+    int64_t totalTime;     ///< Total time taken for the synthesis in us
+    NLOHMANN_DEFINE_TYPE_INTRUSIVE_ONLY_SERIALIZE(Statistics, placementTime,
+                                                  routingTime, totalTime);
+  };
+
+private:
+  /// The statistics collected during the synthesis process
+  Statistics statistics_;
+  /**
+   * @brief Construct a PlaceAndRouteSynthesizer instance with the given
+   * configuration.
+   * @param config is the configuration for the placer and router.
+   * @param architecture is the architecture for which the layout is
+   * synthesized.
+   */
+  PlaceAndRouteSynthesizer(const Architecture& architecture,
+                           const Config& config)
+      : Placer(architecture, config.placerConfig),
+        Router(architecture, config.routerConfig) {}
+
+  /**
+   * @brief Construct a PlaceAndRouteSynthesizer instance with default
+   * configuration.
+   * @param architecture is the architecture for which the layout is
+   * synthesized.
+   */
+  explicit PlaceAndRouteSynthesizer(const Architecture& architecture)
+      : PlaceAndRouteSynthesizer(architecture, Config{}) {}
+
+public:
+  [[nodiscard]] auto synthesize(
+      size_t nQubits, const std::vector<TwoQubitGateLayer>& twoQubitGateLayers,
+      const std::vector<std::unordered_set<qc::Qubit>>& reuseQubits) -> Layout {
+    const auto& placementStart = std::chrono::system_clock::now();
+    const auto& placement =
+        SELF.place(nQubits, twoQubitGateLayers, reuseQubits);
+    statistics_.placementTime =
+        std::chrono::duration_cast<std::chrono::microseconds>(
+            std::chrono::system_clock::now() - placementStart)
+            .count();
+    SPDLOG_INFO("Time for placement: {}us", statistics_.placementTime);
+
+    const auto& routingStart = std::chrono::system_clock::now();
+    const auto& routing = SELF.route(placement);
+    statistics_.routingTime =
+        std::chrono::duration_cast<std::chrono::microseconds>(
+            std::chrono::system_clock::now() - routingStart)
+            .count();
+    SPDLOG_INFO("Time for routing: {}us", statistics_.routingTime);
+
+    statistics_.totalTime =
+        std::chrono::duration_cast<std::chrono::microseconds>(
+            std::chrono::system_clock::now() - placementStart)
+            .count();
+    SPDLOG_INFO("Total time: {}us", statistics_.totalTime);
+
```

Comment:
Also just a nitpick, but you could reduce the number of calls to the system clock from 5 down to 3 by simply storing the timestamp of the completions.